axelrod/tests/unit/test_filters.py

```diff
@@ -0,0 +1,147 @@
+            self.TestStrategy, 'memory_depth', larger, operator.ge))
+
+    def test_list_filter(self):
+        self.assertTrue(passes_in_list_filter(
+            self.TestStrategy, 'makes_use_of', 'game'))
+        self.assertTrue(passes_in_list_filter(
+            self.TestStrategy, 'makes_use_of', 'length'))
+        self.assertFalse(passes_in_list_filter(
+            self.TestStrategy, 'makes_use_of', 'test'))
+
+    @given(
+        smaller=integers(min_value=0, max_value=9),
+        larger=integers(min_value=11, max_value=100),
+    )
+    @example(smaller=0, larger=float('inf'))
+    def test_passes_filterset(self, smaller, larger):
+
+        full_passing_filterset = {
+            'stochastic': True,
+            'inspects_source': False,
+            'min_memory_depth': smaller,
+            'max_memory_depth': larger,
+            'makes_use_of': 'length'
+        }
+
+        sparse_passing_filterset = {
+            'stochastic': True,
+            'inspects_source': False,
+            'makes_use_of': 'length'
+        }
+
+        full_failing_filterset = {
+            'stochastic': False,
+            'inspects_source': False,
+            'min_memory_depth': smaller,
+            'max_memory_depth': larger,
+            'makes_use_of': 'length'
+        }
+
+        sparse_failing_filterset = {
+            'stochastic': False,
+            'inspects_source': False,
+            'min_memory_depth': smaller,
+        }
+
+        self.assertTrue(passes_filterset(
+            self.TestStrategy, full_passing_filterset))
+        self.assertTrue(passes_filterset(
+            self.TestStrategy, sparse_passing_filterset))
+        self.assertFalse(passes_filterset(
+            self.TestStrategy, full_failing_filterset))
+        self.assertFalse(passes_filterset(
+            self.TestStrategy, sparse_failing_filterset))
+
+    def test_filtered_strategies(self):
+
+        class StochasticTestStrategy(object):
+            classifier = {
+                'stochastic': True,
+                'memory_depth': float('inf'),
+                'makes_use_of': []
+            }
+
+        class MemoryDepth2TestStrategy(object):
+            classifier = {
+                'stochastic': False,
+                'memory_depth': 2,
+                'makes_use_of': []
+            }
+
+        class UsesLengthTestStrategy(object):
+            classifier = {
+                'stochastic': True,
+                'memory_depth': float('inf'),
+                'makes_use_of': ['length']
+            }
+
+        strategies = [
+            StochasticTestStrategy,
+            MemoryDepth2TestStrategy,
+            UsesLengthTestStrategy
+        ]
+
+        stochastic_filterset = {
+            'stochastic': True
+        }
+
+        deterministic_filterset = {
+            'stochastic': False
+        }
+
+        uses_length_filterset = {
+            'stochastic': True,
+            'makes_use_of': 'length'
```

Comment:
I'll amend the list filter so that it takes another list as input and checks that all items from the given list are present in the strategy's classifier. I'll work out how to handle passing that list in the API.